docs/_howto/stub-dependency.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: page
+title: How to stub a dependency of a module?
+---
+
+Sinon is simply a stubbing library, not a module interception library. Stubbing dependencies is highly dependant on your enviroment and the implementation. For Node environments, we usually recommend solutions targetting [link seams](./link-seams-commonjs) or explicit dependency injection. Though in some simple cases, you can get away with just using Sinon by modifying the module exports of the dependency.
+
+To stub a dependency (imported module) of a module under test you have to import it explicitly in your test and stub the desired method. For the mocking to work, the mocked method cannot be deconstructed, neither in the module under test nor in the test.
```

Comment:
I'd replace "mocking" with "stubbing" and "mocked" with "stubbed", since we actually differ "mocks" from "stubs" in our docs and API. Otherwise, people might get a bit confused.

Also, replace "deconstructed" with "destructured", using a link to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment). That's assuming I understand it's actually destructuring you refer to? As in
```javascript
const {a} = require('/module-with-a-export');
```

Reply:
@fatso83 Makes sense, yes you are right about destructed, and the link to MDN makes this more clear. Good feedback :)

I've updated the code accordingly.

BTW, I see there have already been discussions about this topic.
https://github.com/sinonjs/sinon/issues/1538

Hopefully this documentation we are adding will help developers having problems with stubbing dependencies this way :)